manifests/params.pp

```diff
@@ -166,18 +166,13 @@
       $bindir                 = pick($bindir, "/usr/lib/postgresql/${version}/bin")
       $datadir                = pick($datadir, "/var/lib/postgresql/${version}/main")
       $confdir                = pick($confdir, "/etc/postgresql/${version}/main")
-      if $facts['os']['name'] == 'Debian' and versioncmp($facts['os']['release']['major'], '8') >= 0 {
-        # Jessie uses systemd
-        $service_status = pick($service_status, "/usr/sbin/service ${service_name}@*-main status")
-      } elsif $facts['os']['name'] == 'Ubuntu' and versioncmp($facts['os']['release']['major'], '18.04') >= 0 {
-        $service_status = pick($service_status, "/usr/sbin/service ${service_name}@*-main status")
-      } elsif $facts['os']['name'] == 'Ubuntu' and versioncmp($facts['os']['release']['major'], '15.04') >= 0 {
-        # Ubuntu releases since vivid use systemd
-        $service_status = pick($service_status, "/usr/sbin/service ${service_name} status")
+      if pick($service_provider, $facts['service_provider']) == 'systemd' {
+        $service_reload = "systemctl reload ${service_name}"
+        $service_status = pick($service_status, "systemctl status ${service_name}")
       } else {
-        $service_status = pick($service_status, "/etc/init.d/${service_name} status | /bin/egrep -q 'Running clusters: .+|online'")
+        $service_reload = "service ${service_name} reload"
+        $service_status = pick($service_status, "service ${service_name} status")
```

Comment:
Oh, I think these paths need to be absolute.

Reply:
Oh yeah, I cheated a bit, as it moved from `/bin/` to `/usr/bin/` or the other way around, but I think both work anyway (as does omitting the path, but I agree, having full path is better :) ).

Reply:
I'd be OK with that (I've seen that Puppet does apply `$PATH` to it, unlike `exec`), but then the unit tests should match.